src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -112,6 +114,19 @@ FinishRemoteTransactionBegin(struct MultiConnection *connection)
 	{
 		Assert(PQtransactionStatus(connection->pgConn) == PQTRANS_INTRANS);
 	}
+
+	if (!transaction->transactionFailed)
```

Comment:
Move to statement built in `StartRemoteTransactionBegin()`, to avoid additional roundtrips.